lib/compiler/src/engine/trap/frame_info.rs

```diff
@@ -54,7 +54,7 @@ pub struct GlobalFrameInfoRegistration {
 struct ModuleInfoFrameInfo {
     start: usize,
     functions: BTreeMap<usize, FunctionInfo>,
-    module: ModuleInfo,
+    module: Arc<ModuleInfo>,
```

Comment:
Why is this Arc needed here?

Reply:
Basically, before we were cloning the `ModuleInfo` in multiple places, but that's no longer needed.

By having an Arc, we allow it to be cloned *only* when we set up the module frame info registration (is needed, as otherwise the lifetimes for accessing a pointer will make our life a bit of a hell there), so Arc was the simplest solution